### ISSUE TITLE
Close method not working in 8.13.0

### ISSUE THREAD

**Issue by matheusfig90** [sitespeedio/sitespeed.io#2433]
Hi everyone,

My team is developing a plugin to expose sitespeed metrics in Prometheus. 

An example of code:

```
close(options, errors) {
  return new Promise(resolve => {
    // Sending metrics
    resolve()
  })
}
```

Now, we are using a version `7.7.3` and works very well! But, on upgrade to `8.13.0` doesn't work :(

Anyone has this problem?

We are running in MacOS Mojave (Docker version 18.09.2, build 6247962).

Best Regards,
Matheus

**Comment by soulgalore** [comment]
Hi @matheusfig90 thanks for creating the report, I think it's bug somewhere, let me have a look later tonight. There was another issue just a couple of days ago with almost the same problem so my guess is that I somehow introduced it unintentionally ... however it should be easy to fix when I find it :) 

I'll update the issue ASAP when I have time to have a look.

Best
Peter

**Comment by soulgalore** [comment]
Hi again @matheusfig90 so there was a bug introduced in 8.x (probably 8.0= where we didn't wait on all close methods to finished before moving on. I've fixed that and will release it later tonight (in about 4-5 hours). The reason we haven't noticed is that we moved on to act on message instead of using open close. You can read more about the messages in https://www.sitespeed.io/documentation/sitespeed.io/plugins/#important-messages. So instead on acting on close we mostly act on messages pass through the queue.

Please get back when you tried the new version so I know it really worked for you. 

Best
Peter

**Comment by soulgalore** [comment]
Lets keep it open until its verified.

**Comment by soulgalore** [comment]
8.14.0 is there now, let me know @matheusfig90 when you had time to test it out!

**Comment by matheusfig90** [comment]
@soulgalore man, you are the best! 

it's working now!
thank you very much!

**Comment by soulgalore** [comment]
Great! Well, it was me that introduced the bug so I should fix that :)
